### PR TITLE
feat: remove dependency on OpenPolicyAgent_jll

### LINKED ISSUE #15: Decouple package from OPA jll?
It may be good to decouple the package from tight dependency of `OpenPolicyAgent_jll`. The package allows caller to specify an `opa` executable. Not having a tight dependency will allow calling code to avoid installing the jll at all, and not be affected if the jll is not updated/outdated.

### PR DESCRIPTION
**Pull request by tanmaykm**
Removing hard dependency on `OpenPolicyAgent_jll`. Since we allow caller to specify an opa executable, not having a tight dependency will allow calling code to avoid installing the jll at all. So that it will not be affected if the jll is not updated/outdated.

fixes: #15

### CODE REVIEW

**Comment by tanmaykm** [comment]
Updated to require the `opa` executable command to be provided explicitly. It won't be picked up from `PATH` env anymore. Ref: https://github.com/JuliaComputing/OpenPolicyAgent.jl/issues/15#issuecomment-2441139110

File: specs/cli/gencli.jl [hunk 1]
```diff
@@ -15,7 +13,7 @@ CommandLine execution context.
 `pipelineopts`: keyword arguments that should be used to further customize the `pipeline` creation
 \"\"\"
 Base.@kwdef struct CommandLine
-    exec::Base.Function = OpenPolicyAgent_jll.opa
+    exec::Base.Function = ()->`opa`
```
Comment (mortenpi) [inline specs/cli/gencli.jl]: This needs an update maybe? It seems to be inconsistent with `cli.jl`?

File: src/cli/cli.jl [hunk 2]
```diff
@@ -9,22 +9,27 @@ module CLI
...
-    exec::Base.Function = OpenPolicyAgent_jll.opa
+    exec::Base.Function = ()->``
     cmdopts::OptsType = OptsType()
     pipelineopts::OptsType = OptsType()
     runopts::OptsType = OptsType()
 end
 
+function CommandLine(exec::Base.Function; kwargs...)
+    return CommandLine(; exec=exec, kwargs...)
+end
```
Comment (mortenpi) [inline src/cli/cli.jl]: Do you expect the users to use the 4-keyword constructor? If not, then this should probably be an inner constructor? Defaulting to an empty command seems a bit weird. I.e. something like

```julia
struct CommandLine
    exec::Base.Function
    cmdopts::OptsType
    pipelineopts::OptsType
    runopts::OptsType

    function CommandLine(exec::Base.Function; cmdopts::OptsType = OptsType(), pipelineopts::OptsType = OptsType(), runopts::OptsType = OptsType())
        return new(exec, cmdopts, pipelineopts, runopts)
    end
end
```
?

File: src/server/server.jl [hunk 3]
```diff
@@ -17,6 +19,7 @@ const DEFAULT_PORT = 8181
 A server that is monitored and restarted if it dies.
 
 Arguments:
+- `cmdline`: The `CLI.CommandLine` instrance that wraps an opa executable.
```
Comment (mortenpi) [inline src/server/server.jl]:
```suggestion
- `cmdline`: The `CLI.CommandLine` instance that wraps an opa executable.
```

File: src/server/server.jl [hunk 4]
```diff
@@ -37,16 +40,14 @@ struct MonitoredOPAServer
     stopped::Ref{Bool}
     lck::ReentrantLock
 
-    function MonitoredOPAServer(configfile::String;
+    function MonitoredOPAServer(
+        cmdline::CommandLine,
```
Comment (mortenpi) [inline src/server/server.jl:44]: Bikeshedding a bit, but could maybe have a convenience outer constructor

```julia
MonitoredOPAServer(exec::Function, configfile::String; kwargs...)
= MonitoredOPAServer(CommandLine(exec), configfile; kwargs...)

# Use as:
MonitoredOPAServer(OpenPolicyAgent_jll.opa, ...)
```

Users generally care about the `MonitoredOPAServer` object, and not the `CommandLine` object, right?

File: specs/cli/gencli.jl (hunk 1 above)
Comment (tanmaykm) [reply specs/cli/gencli.jl]: Thanks for catching. Done in https://github.com/JuliaComputing/OpenPolicyAgent.jl/pull/17/commits/55c894db20ee7abc2c4cb8e222473ee743095a17

File: src/cli/cli.jl (hunk 2 above)
Comment (tanmaykm) [reply src/cli/cli.jl]: Yes... let's change it

Comment (tanmaykm) [reply src/cli/cli.jl]: Done in https://github.com/JuliaComputing/OpenPolicyAgent.jl/pull/17/commits/7f777bd21ddb246657d3f21db7458b46e5f4ee0b

File: src/server/server.jl (hunk 4 above)
Comment (tanmaykm) [reply src/server/server.jl:44]: Done in https://github.com/JuliaComputing/OpenPolicyAgent.jl/pull/17/commits/96120546b8d970899c27f91f957c371ec3d554e6

**Review by mortenpi — APPROVED** [review]
LGTM!

**Pull request merged**